app/src/main/java/com/monstarlab/features/auth/domain/AuthRepository.kt

```diff
@@ -1,15 +1,16 @@
 package com.monstarlab.features.auth.domain
 
+import com.monstarlab.features.auth.AuthTokenMapper
 import com.monstarlab.features.auth.data.api.AuthApi
 import com.monstarlab.features.auth.domain.models.AuthToken
 import javax.inject.Inject
 
 class AuthRepository @Inject constructor(
     private val api: AuthApi,
+    private val authTokenMapper: AuthTokenMapper
```

Comment:
With the Mappers discussion aside, this injection doesn't bring any benefits since you inject an implementation and not an Interface. And since its a Singleton it can just be an `object`

Reply:
Great observation! The purpose of this mapper class is to provide mapping functionality between two different object types, making the code more modular, testable, and maintainable. This approach is not primarily intended for dependency injection as the mapper interface is generic.

However, creating a singleton using an `object` keyword has a downside - it is not easy to mock the `AuthTokenMapper` during unit testing. If you need to test classes that depend on the `AuthTokenMapper`, using dependency injection to provide a mock implementation may be a better approach.

Reply:
Even with the mappers as is, I don't see a reason why mock such a simple thing if you can use its Implementation directly. It doesn't have any dependencies on its own + there is no test for the Mapper Implementation. So if someone changes the Mapper implementation and makes a mistake - The test will pass anyways